tests/unittests/HyphenTest.cpp

```diff
@@ -0,0 +1,322 @@
+    "in-crease",
+    "in-di-vi-du-al",
+    "it-self",
+    "lan-guage",
+    "le-gal",
+    "lit-tle",
+    "lo-cal",
+    "ma-jo-ri-ty",
+    "ma-te-rial",
+    "may-be",
+    "me-di-cal",
+    "meet-ing",
+    "men-tion",
+    "mid-dle",
+    "na-tion",
+    "na-tional",
+    "oc-cur",
+    "of-fi-cer",
+    "par-tic-u-lar-ly",
+    "pat-tern",
+    "pe-riod",
+    "phy-si-cal",
+    "po-si-tion",
+    "pol-icy",
+    "pos-si-ble",
+    "pre-vent",
+    "pres-sure",
+    "pro-per-ty",
+    "pur-pose",
+    "re-cog-nize",
+    "re-gion",
+    "re-la-tion-ship",
+    "re-main",
+    "re-sponse",
+    "re-sult",
+    "rea-son",
+    "sea-son",
+    "sex-ual",
+    "si-mi-lar",
+    "sig-ni-fi-cant",
+    "sim-ple",
+    "sud-den-ly",
+    "sum-mer",
+    "thou-sand",
+    "to-day",
+    "train-ing",
+    "treat-ment",
+    "va-ri-ous",
+    "value",
+    "vi-o-lence",
+};
+
+TEST(HyphenTest, network) {
+  // Convert the training data to word windows and labels.
+  vector<string> words;
+  vector<bool> hyphens;
+  for (auto *hword : TrainingData) {
+    dehyphenate(hword, words, hyphens);
+  }
+
+  // This depends on the training data, of course.
+  const size_t numSamples = 566;
+  ASSERT_EQ(hyphens.size(), numSamples);
+  ASSERT_EQ(words.size(), numSamples);
+
+  // Randomly shuffle the training data.
+  // This is required for stochastic gradient descent training.
+  for (size_t i = 0; i != numSamples; i++) {
+    size_t j = nextRandInt(0, numSamples - 1);
+    std::swap(words[i], words[j]);
+    std::swap(hyphens[i], hyphens[j]);
+  }
+
+  // Convert words and hyphens to a tensor representation.
+  Tensor inputs(ElemKind::FloatTy, {numSamples, 6, 27});
+  Tensor outputs(ElemKind::FloatTy, {numSamples, 1});
+  auto inputHandle = inputs.getHandle<float>();
+  auto outputHandle = outputs.getHandle<float>();
+  for (size_t i = 0; i != numSamples; i++) {
+    mapLetterWindow(words[i], i, inputHandle);
+    outputHandle.at({i, 0}) = hyphens[i];
+  }
+
+  // Now build the network:
+  //
+  // Input: 6 x 27.
+  // Hidden: 16 fully connected, RELU output.
+  // Output: 1 fully connected, sigmoid output for logistic regression.
+  const size_t batchSize = numSamples;
+  ExecutionEngine EE(BackendKind::CPU);
+  auto &mod = EE.getModule();
+  Function *Infer = mod.createFunction("infer");
+
+  auto *InputVar =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 6, 27}, "input",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+  auto *OutputVar =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "output",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *HiddenFC = Infer->createFullyConnected("hidden_fc", InputVar, 16);
+  auto *Hidden = Infer->createRELU("hidden", HiddenFC);
+  auto *OutputFC = Infer->createFullyConnected("output_fc", Hidden, 1);
+  auto *Sig = Infer->createSigmoid("sig", OutputFC);
+  auto *Output = Infer->createRegression("regr", Sig, OutputVar);
```

Comment:
Regression in Glow means linear regression. The net will minimize Total Squared Error, while we really have classification task. Let's use `Softmax`

Reply:
Thanks. `Softmax` requires two outputs for a binary classification, right? Is there no way of getting a single yes/no answer with logistic regression from one output neuron?

Reply:
Yes, it will require 2 neurons since we have 2 classes. Maybe change last FC to depth=2